src/GafferImage/ImageStats.cpp

```diff
@@ -198,98 +222,219 @@ const DeepState *ImageStats::deepState() const
 void ImageStats::affects( const Gaffer::Plug *input, AffectedPlugsContainer &outputs ) const
 {
 	ComputeNode::affects( input, outputs );
+
 	if(
 		input == flattenedInPlug()->dataWindowPlug() ||
-		input == flattenedInPlug()->channelNamesPlug() ||
 		input == flattenedInPlug()->channelDataPlug() ||
 		input == channelsPlug() ||
 		areaPlug()->isAncestorOf( input )
 	)
+	{
+		outputs.push_back( tileStatsPlug() );
+	}
+	else if(
+		input == tileStatsPlug() ||
+		input == flattenedInPlug()->dataWindowPlug() ||
+		areaPlug()->isAncestorOf( input )
+	)
+	{
+		outputs.push_back( allStatsPlug() );
+	}
+	else if(
+		input == allStatsPlug() ||
+		input == flattenedInPlug()->channelNamesPlug()
+	)
```

Comment:
We still need `|| input == channelsPlug()` here, as the compute pulls on it indirectly via the `channelName()` method. Unless it hides significant complexity, I'm not a fan of "helpers" like `channelName()` when they hide plug accesses because they encourage bugs like this. I think `colorIndex()` and `channelName()` could be collapsed into a single static helper method which is passed plugs and/or values directly so it doesn't hide any plug accesses.